### PR TITLE
doc update for only applying the patches when using cuda

### PR DESCRIPTION
**Pull request by quasiben**
Follow up for https://github.com/rapidsai/ucx-py/issues/628#issuecomment-702415714

cc @pentschev

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -87,6 +87,9 @@ UCX
     ../contrib/configure-devel --prefix=$CONDA_PREFIX --with-cuda=$CUDA_HOME --enable-mt CPPFLAGS="-I/$CUDA_HOME/include"
     make -j install
 
+.. note::
+    If you're running on a machine without CUDA then you can skip applying the patches
```
Comment (pentschev) [inline docs/source/install.rst:91]:
```suggestion
    If you're running on a machine without CUDA then you _must NOT_ apply the patches.
```

**Review by pentschev — CHANGES_REQUESTED** [review]
I would make the comment more emphatic.

**Review by pentschev — APPROVED** [review]
LGTM, thanks @quasiben .

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Ben! 😄

**Pull request merged**